Implement rest api save user permissions endpoint [#33393]

hassandotcms opened the pull request:
### Proposed Changes

- Implemented REST API to save/update user permissions on specific assets (hosts/folders)
- Supports multiple permission scopes (INDIVIDUAL, HOST, FOLDER) and levels (READ, WRITE, PUBLISH, EDIT_PERMISSIONS, CAN_ADD_CHILDREN)
- Supports cascading permissions to child assets
- Added integration tests covering success scenarios, validation, and security checks

fixes #33393

This PR fixes: #33393

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

- add new response view class.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)
- implement helper methods for user permission rest api.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)
 - get api implementation to return user specific permissions by id/email, by hosts/folder.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

- integration test added to verify the functionality of get user permissions rest api.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

update api docs.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

add postman test.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

fix integration tests, response type error.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

support api classes to be available for injection via CDI.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

enable CDI for userPermissionHelper, to enable injection.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

 fix integ tests

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

  - Return specific response type instead of generic Response
  - Add better documentation and error messages
  - Split user loading into separate helper method

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

 refactoring and cleanup.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33345-implement-rest-api-get-user-permissions-endpoint

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions by user. (#33345)

move helper method to helper class.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

 add view class for permission metadata.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

implement rest api for permission metadata.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

add javadoc, refactoring in helper methods to make them reusable and update api docs.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

 fix constructor init issue.

hassandotcms committed:
modernize(users portlet): implement rest api to fetch permissions metadata. (#33394)

 add postman test for permission metadata api

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(users-portlet) #33394

 - Create PermissionMetadata, UserPermissions, and UserPermissionAsset beans for type-safe API responses
 - Update permission endpoints to return ResponseEntityView wrappers instead of generic Response objects
 - Replace Map-based data structures with beans

hassandotcms committed:
modernize(users-portlet): rest api for saving permissions [#33393]

- created rest resource endpoint to save/update user permissions.
- helper methods to validate transform permission data

hassandotcms committed:
modernize(users-portlet): rest api for saving permissions [#33393]

- write integration tests to validate the user permissions.
  success/validation scenarios.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
Merge branch '33394-implement-rest-api-get-permission-metadata' into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(users-portlet): rest api for permission metadata #33394

update auth for permission api.

hassandotcms committed:
Merge branch 'main' into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(users-portlet): rest api to save user permissions #33393

 1. Change List→Set for permission levels to prevent duplicates (PermissionMetadataView, SaveUserPermissionsForm, UserPermissionAsset, UserPermissionHelper)
  2. Rename PermissionMetadata→PermissionMetadataView following dotCMS naming conventions for REST views
  3. Replace switch statement with Function map in convertPermissionNamesToBits(), rename systemUser→user parameter

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
refactor(permissions): align metadata API with specification #33394

- Change endpoint path from /metadata to / (root collection path)
  - Update permission scope terminology from STRUCTURE to CONTENT_TYPE
  - Add 403 Forbidden response documentation to OpenAPI annotations
  - Update Postman collection to use new endpoint path

hassandotcms committed:
api doc update

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
Merge branch '33394-implement-rest-api-get-permission-metadata' into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(permissions): refactor user permissions API #33393

  Added PermissionResourceIntegrationTest with tests for GET/PUT endpoints.
  Updated PermissionResourceIntegrationTest and Postman collection to use new paths.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
reorganize files and sync with main. #33394

hassandotcms committed:
modernize(permissions): refactor utils. #33394

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
remove unrelated user package changes #33394

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(permissions): rest api to update user permisisons #33393

sync with main and refactoring of file organization and tests.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
sync main

hassandotcms committed:
modernize(permissions): rest api to update user permissions (#33393)

fix failing tests.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(permissions): rest api to update user permissions [#33393]

fix integration tests.

hassandotcms committed:
Merge branch 'main' into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(permissions): rest api to save user permissions [#33393]

 1. Updated @author annotations (5 files)
  - Changed @author dotCMS → @author hassandotcms in:
    - UserPermissionAssetView.java
    - ResponseEntitySaveUserPermissionsView.java
    - PermissionSaveHelper.java
    - SaveUserPermissionsForm.java
    - SaveUserPermissionsView.java

  2. Added structured test javadocs (10 methods)
  - Added <ul><li><b>Method to test:</b>, <b>Given Scenario:</b>, <b>Expected Result:</b> format to all test methods in PermissionResourceIntegrationTest.java

  3. Added test to CI suite
  - Added import + PermissionResourceIntegrationTest.class to MainSuite2b.java

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms commented on dotcms-integration/src/test/java/com/dotcms/rest/api/v1/system/permission/PermissionResourceIntegrationTest.java:
```diff
@@ -0,0 +1,441 @@
+    static Host permissionTestHost;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
resolved

hassandotcms commented on dotcms-integration/src/test/java/com/dotcms/rest/api/v1/system/permission/PermissionResourceIntegrationTest.java:
```diff
@@ -0,0 +1,441 @@
+
+    // ==================== PUT Permission Tests ====================
+
+    @Test
```
done

fabrizzio-dotCMS commented on dotCMS/src/main/java/com/dotcms/rest/api/v1/system/permission/PermissionMetadataView.java:
```diff
@@ -0,0 +1,62 @@
+package com.dotcms.rest.api.v1.system.permission;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
```
Again, review our uses of the immutables library. But this is fine

hassandotcms committed:
modernize(permission): rest api to get permission metadata [#33394]

- use immutables for views. AbstractPermissionMetadataView

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33394-implement-rest-api-get-permission-metadata

hassandotcms committed:
modernize(permissions): rest api to save user permissions (#33393)

- fix order of tests in mainsuite2b

hassandotcms committed:
modernize(permissions): api to get permissions metadata [#33394]

- use immutables
- move relevant methods to PermissionAPI

hassandotcms committed:
Merge branch '33394-implement-rest-api-get-permission-metadata' into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(permissions): rest api to save user permissions (#33393)
 - Added PermissionAPI.Scope enum + refactored PermissionSaveHelper to use it
 - Converted UserPermissionAssetView to Immutables (deleted class → new AbstractUserPermissionAssetView interface)

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(permission): fix integration test to use proper imports. #33393

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint

hassandotcms committed:
modernize(permissions): fix failing integration tests, add updateUserPermissions method. #33393

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 33393-implement-rest-api-save-user-permissions-endpoint
